main/modes/mode_diceroller.c

```diff
@@ -242,310 +342,234 @@ void doStateMachine(int64_t elapsedUs)
+        totalStr,
+        histX - textWidth(&diceRoller->ibm_vga8,totalStr)/2,
+        histY
+    );
+
+    for(int i = 0; i < diceRoller->histSize; i++)
+    {
+        snprintf(totalStr,sizeof(totalStr),"%dd%d: %d",diceRoller->histCounts[i],diceRoller->histSides[i],diceRoller->histTotals[i]);
+        drawText(
+            diceRoller->disp,
+            &diceRoller->ibm_vga8, histTextColor,
+            totalStr,
+            histX - textWidth(&diceRoller->ibm_vga8,totalStr)/2,
+            histY + (i+1)*histYEntryOffset
+        );
+    }
+}
 
-                int64_t periodUs = 10000000;
-                diceRoller->timeAngle += elapsedUs%periodUs/(double)periodUs*360;
+void addTotalToHistory()
+{
+    if(diceRoller->histSize < DR_MAXHIST) //S
+    {
+        
+        int size = diceRoller->histSize;
+        for(int i = 0; i < size; i++)
+        {
+            
+                diceRoller->histTotals[size-i] = diceRoller->histTotals[size-i-1]; //Shift vals to right
+                diceRoller->histCounts[size-i] = diceRoller->histCounts[size-i-1];
+                diceRoller->histSides[size-i] = diceRoller->histSides[size-i-1];
+            
+        }
+        diceRoller->histTotals[0] = diceRoller->rollTotal;
+        diceRoller->histCounts[0] = diceRoller->rollSize;
+        diceRoller->histSides[0] = diceRoller->rollSides;
+        diceRoller->histSize += 1;
+    }
+    else //shift out last value;
+    {
+        int size = diceRoller->histSize;
+        for(int i = 0; i < size; i++)
+        {
+            if(i < size-1)
+            {
+                diceRoller->histTotals[size-1-i] = diceRoller->histTotals[size-2-i]; //Shift vals to right
+                diceRoller->histCounts[size-1-i] = diceRoller->histCounts[size-2-i];
+                diceRoller->histSides[size-1-i] = diceRoller->histSides[size-2-i];
+            }
+            else
+            {
+                diceRoller->histTotals[0] = diceRoller->rollTotal;
+                diceRoller->histCounts[0] = diceRoller->rollSize;
+                diceRoller->histSides[0] = diceRoller->rollSides;
+            }
+        }
+    }
 
+}
 
-                //Cool Nested Spinny Polygons
-                //drawRegularPolygon(diceRoller->disp->w/2, diceRoller->disp->h/2, 3, 3*diceRoller->timeAngle, 50, c555, 0);
-                //drawRegularPolygon(diceRoller->disp->w/2, diceRoller->disp->h/2, 4, 4*diceRoller->timeAngle, 60, c555, 0);
-                //drawRegularPolygon(diceRoller->disp->w/2, diceRoller->disp->h/2, 5, 5*diceRoller->timeAngle, 70, c555, 0);
-                //drawRegularPolygon(diceRoller->disp->w/2, diceRoller->disp->h/2, 6, 6*diceRoller->timeAngle, 80, c555, 0);
-                if(rollAnimationTimeUs > rollAnimationPeriod)
-                {
-                    diceRoller->state = DR_SHOWROLL;
-                }
-                break;
-        }
-        default:
+void dbgPrintHist()
+{
+    printf("History:");

```

Comment:
Try to use `ESP_LOGD()` instead of `printf()`. There's no way for you to know that, but it's easier to disable debug for production with that macro.

Reply:
Fixed